components/campuses/Campuses.jsx

```diff
@@ -0,0 +1,157 @@
+import * as React from 'react';
+import { Button, CircularProgress, Grid, withStyles } from '@material-ui/core';
+import { db } from '../../src/firebase/firebaseSetup.js';
+import { getAllFromFirestore, uploadImage } from '../Helpers';
+import Campus from './CampusesCard';
+import EditForm from './CampusesEditForm';
+
+const styles = style => ({
+    progress: {
+      margin: style.spacing(2),
+    },
+    formControl: {
+      margin: style.spacing(1),
+      minWidth: 120,
+    },
+    newCampusbutton: {
+      float: 'right'
+    }
+});
+
+// This component is the base of Campuses tab
+class Campuses extends React.Component {
+    constructor (props) {
+      super(props);
+      this.state = {
+        campuses: [],
+        loading: true,
+        openForm: false,
+      };
+
+      this.getCampuses = this.getCampuses.bind(this);
+      this.handleForm = this.handleForm.bind(this);
+      this.removeCampus = this.removeCampus.bind(this);
+    }
+
+    componentDidMount() {
+      this.getCampuses();
+    }
+
+    getCampuses = () => {
+        // get all campuses
+        var campuses = getAllFromFirestore('campuses');
+        
+        // combine collections to pass to Campus Cards
+        Promise.all([campuses]).then((data) => {
+          var campuses = data[0];
+          this.setState({
+            campuses,
+            loading: false
+          });
+        });
+    }
+
+    // uploads campus and re-renders campus view
+    uploadAndAddCampus = (campus, url) => {
+      var newCampuses = this.state.campuses;
+      db.collection("campuses").add({
+        location: {
+          city: campus.location.city,
+          country: campus.location.country,
+          number: campus.location.number,
+          state: campus.location.state,
+          street1: campus.location.street1,
+          street2: campus.location.street2
+        },
+        imageLink: url,
+        name: campus.name
+      }).then((campusCallback) => {
+        // id needed for Firestore to update, imageLink retrieves image
+        campus.id = campusCallback.id;
+        campus.imageLink = url;
+        newCampuses.push(campus);
+        this.setState({
+          campuses: newCampuses
+        });
+      });
+    }
+
+      /* Callback to remove the Campuse from those Campuses 
+      displayed based on index */ 
+    removeCampus = (id) => {
+      db.collection("campuses").doc(id).delete().
+      then(() => {
+          let campuses = this.state.campuses;
+          // look for campuse to remove
+          for (let i = 0; i < campuses.length; i++) {

```

Comment:
This seems like the perfect use for a filter function.
``` js
let campus = this.state.campuses.filter(campus => campus.id !== id);
```

Reply:
Good catch!